usr/libexec/greenboot/greenboot-grub2-set-counter

```diff
@@ -14,7 +14,31 @@ elif [ -n "$GREENBOOT_MAX_BOOT_ATTEMPTS" ]; then
 else
     max_boot_attempts=3 # default to 3 attempts
 fi
-grub2-editenv - set boot_counter="$max_boot_attempts"
-grub2-editenv - set boot_success=0
+
+# Remount /boot as read-only if it was mounted as read-write
+if grep -q " /boot .* rw," /proc/mounts; then
+    mount -o remount,ro /boot || exit 1
+fi
+
+if ! /usr/bin/grub2-editenv - set boot_counter="$max_boot_attempts"; then
+ # If the first command fails, remount /boot as read-only and exit with failure
+    if grep -q " /boot .* rw," /proc/mounts; then
+        mount -o remount,ro /boot || exit 1
+    fi
+exit 1
```

Comment:
missing indentation

Reply:
also maybe it makes sense to exit with different values (instead of `exit 1` everywhere) in order to be able to tell the different failure modes apart.

Reply:
changed all `mount` error to 5